json.go

```diff
@@ -75,6 +75,11 @@ const (
 
 	// DisallowUnknownFields returns strict errors if data contains unknown fields when decoding into typed structs
 	DisallowUnknownFields StrictOption = 2
+
```

Comment:
rather than making a new decode option and truncating the non-path information in the returned error, it might be simpler to make the returned strict errors conform to an interface so the field path can be retrieved/set:

```
FieldPath() string
SetFieldPath(string)
```

then consumers of the strict errors that need to get field path info could extract it, and consumers that need to prepend additional field paths could do so as well